From: Mark Fasheh <mfasheh@suse.de>
Date: Thu, 8 Sep 2011 17:14:32 -0700
Subject: btrfs: Don't BUG_ON insert errors in btrfs_alloc_dev_extent()
Patch-mainline: 3.4-rc1
Git-commit: 2cdcecbc153c222fae1be6f8ddb320b29e3a5200

The only caller of btrfs_alloc_dev_extent() is __btrfs_alloc_chunk() which
already bugs on any error returned. We can remove the BUG_ON's in
btrfs_alloc_dev_extent() then since __btrfs_alloc_chunk() will "catch" them
anyway.

Signed-off-by: Mark Fasheh <mfasheh@suse.de>
---
 fs/btrfs/volumes.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/fs/btrfs/volumes.c
+++ b/fs/btrfs/volumes.c
@@ -1076,7 +1076,8 @@ int btrfs_alloc_dev_extent(struct btrfs_
 	key.type = BTRFS_DEV_EXTENT_KEY;
 	ret = btrfs_insert_empty_item(trans, root, path, &key,
 				      sizeof(*extent));
-	BUG_ON(ret);
+	if (ret)
+		goto out;
 
 	leaf = path->nodes[0];
 	extent = btrfs_item_ptr(leaf, path->slots[0],
@@ -1091,6 +1092,7 @@ int btrfs_alloc_dev_extent(struct btrfs_
 
 	btrfs_set_dev_extent_length(leaf, extent, num_bytes);
 	btrfs_mark_buffer_dirty(leaf);
+out:
 	btrfs_free_path(path);
 	return ret;
 }
